### PR TITLE
[ao][sparsity] Implemented the step() function

### PR DESCRIPTION
**Pull request by macandro96**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79883
* __->__ #79822
* #79817

The step() function calls the implemented get_schedule_param() and updates
the data_group dictionary of the data sparsifier.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataScheduler```

Differential Revision: [D37358609](https://our.internmc.facebook.com/intern/diff/D37358609)

### CODE REVIEW

File: test/ao/sparsity/test_data_scheduler.py [hunk 1]
```diff
@@ -38,12 +44,14 @@ def _get_data(self):
         ]
         return data_list, defaults, data_with_config
 
-    def _get_sparsifier(self):
+    def _get_sparsifier(self, return_data=False):
```
Comment (jerryzh168) [inline test/ao/sparsity/test_data_scheduler.py:47]: nit: if we pass around data in this api as suggested by the last PR we probably do not need this argument

File: test/ao/sparsity/test_data_scheduler.py [hunk 2]
```diff
@@ -54,13 +62,63 @@ def _get_scheduler(self, sparsifier):
...
+        with warnings.catch_warnings(record=True) as w:
+            sparsifier.step()
+            scheduler.step()
+            # Make sure there is no warning related to the base_data_scheduler
+            for warning in w:
+                fname = warning.filename
+                fname = '/'.join(fname.split('/')[-5:])
+                assert fname != 'torch/ao/sparsity/experimental/scheduler/data_scheduler/base_data_scheduler.py'
+
+    def test_step(self):
```
[the remark this answers is not in the document]
Comment (macandro96) [reply test/ao/sparsity/test_data_scheduler.py:107]: added the test for step_counter inside test_step()

**Review by z-a-f — APPROVED** [review]

**Comment by macandro96** [comment]
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37358609).

**Comment by macandro96** [comment]
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37358609).

**Pull request closed without merging**